fem/tmop.hpp

```diff
@@ -1047,6 +1072,37 @@ class TMOP_Integrator : public NonlinearFormIntegrator
    virtual void AssembleElementGrad(const FiniteElement &el,
                                     ElementTransformation &T,
                                     const Vector &elfun, DenseMatrix &elmat);
+   /// PA extension
+   void ComputeElementTargetsPA(const Vector &x = Vector()) const;
+
+   using NonlinearFormIntegrator::GetGridFunctionEnergyPA;
+   double GetGridFunctionEnergyPA_2D(const Vector&) const;
+   double GetGridFunctionEnergyPA_C0_2D(const Vector&) const;
+   double GetGridFunctionEnergyPA_3D(const Vector&) const;
```

Comment:
Do the helper functions have to be `public`?

Reply:
Yes, unfortunately this is required by `nvcc`.